From 015ef004ed2d4ed858e86080707588ae1d329a1f Mon Sep 17 00:00:00 2001
From: bitthief <bitthief@protonmail.ch>
Date: Wed, 9 Mar 2022 04:56:03 +0200
Subject: [PATCH] qca-nss-ecm: define static values for nf_tcp_check_window

Define static values for nf_ct_tcp_check_window and nf_ct_tcp_be_liberal
on kernel 5.15, to accomodate the newly-introduced sysctl for
nf_ct_tcp_check_window.
---
 frontends/include/ecm_front_end_common.h | 10 ++++++++++
 1 file changed, 10 insertions(+)

--- a/frontends/include/ecm_front_end_common.h
+++ b/frontends/include/ecm_front_end_common.h
@@ -18,6 +18,18 @@
 #include <net/netfilter/nf_conntrack.h>
 #include <net/netfilter/nf_conntrack_acct.h>
 
+#ifdef ECM_OPENWRT_SUPPORT
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 15, 0)
+static int __rcu nf_ct_tcp_no_window_check __attribute__((__unused__)) __read_mostly = 1;
+#else
+static struct net __rcu *net __attribute__((__unused__)) __read_mostly;
+#endif
+#endif /* ECM_OPENWRT_SUPPORT */
+
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 13, 0)
+static int __rcu nf_ct_tcp_be_liberal __attribute__((__unused__)) __read_mostly = 0;
+#endif
+
 #ifdef ECM_FRONT_END_NSS_ENABLE
 #include "ecm_nss_bond_notifier.h"
 #else
